src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -162,34 +162,47 @@ public void configure() {
                 .id("unzip-file")
                 .choice()
                     .when(isZippedFile("zip"))
-                        .split(new ZipSplitter())
+                        .split(new ZipSplitter()).aggregationStrategy(this::calculateICSAggregated)
                         .streaming()
                         .to("direct:calculate")
                     .endChoice()
                     .when(isZippedFile("tar.gz"))
                         .unmarshal().gzip()
-                        .split(new TarSplitter())
-                        .streaming()
-                        .to("direct:calculate")
+                        .split(new TarSplitter()).aggregationStrategy(this::calculateICSAggregated)
+                            .streaming()
+                            .to("direct:calculate")
+                        .end()
                     .endChoice()
                     .otherwise()
                         .to("direct:calculate")
-                .end();
+                        .process(exchange -> exchange.getIn().getHeaders().put("uploadformdata", exchange.getIn().getBody(UploadFormInputDataModel.class)))
```

Comment:
Please make `uploadformdata` a static field like `ANALYSIS_ID` and reference it where needed

Reply:
sure. I dont have a clear criteria to know when it's important and when itt's not. There are several hardcoded texts in the code : identity, internal, user, username, file ...
I will replace the ones accessed in **header** method

Reply:
Maybe we can start with having the headers we add to be managed with static fields like `ANALYSIS_ID` and `USERNAME`.

Reply:
ok -> MA_METADATA, UPLOADFORMDATA